pkg/cloud/vsphere/services/loadbalancer/aws/loadbalancers.go

```diff
@@ -0,0 +1,315 @@
+}
+
+func (svc *Service) reconcileTargetGroup(clusterName string, vpcID string, controlPlaneIPs []string) (*string, error) {
+	describeTargetGroupInput := &elbv2.DescribeTargetGroupsInput{}
+	describeTargetGroupInput.Names = append(describeTargetGroupInput.Names, aws.String(clusterName+"-controlPlane"))
+
+	describeTargetGroupOutput, err := svc.ELB.DescribeTargetGroups(describeTargetGroupInput)
+	desiredTargetGroupInput := &elbv2.CreateTargetGroupInput{
+		Name:                       aws.String(clusterName + "-controlPlane"),
+		HealthCheckEnabled:         aws.Bool(true),
+		HealthCheckIntervalSeconds: aws.Int64(int64(30 * time.Second.Seconds())),
+		HealthCheckTimeoutSeconds:  aws.Int64(int64(10 * time.Second.Seconds())),
+		HealthyThresholdCount:      aws.Int64(int64(3)),
+		UnhealthyThresholdCount:    aws.Int64(int64(3)),
+		Port:                       aws.Int64(int64(6443)),
+		Protocol:                   aws.String(elbv2.ProtocolEnumTcp),
+		TargetType:                 aws.String(string("ip")),
+		VpcId:                      &vpcID,
+	}
+
+	var targetGroupArn *string
+	if err != nil {
+		if IsNotFound(err) {
+			targetGroupOutput, err := svc.ELB.CreateTargetGroup(desiredTargetGroupInput)
+			if err != nil {
+				return nil, err
+			}
+			targetGroupArn = targetGroupOutput.TargetGroups[0].TargetGroupArn
+		} else {
+			return nil, err
+		}
+	}
+
+	actualTargetGroupInput := convertTargetGroupOutputToInput(describeTargetGroupOutput.TargetGroups[0])
+	if !reflect.DeepEqual(actualTargetGroupInput, desiredTargetGroupInput) {
+		modifyTargetGroup := &elbv2.ModifyTargetGroupInput{
+			HealthCheckEnabled:         desiredTargetGroupInput.HealthCheckEnabled,
+			HealthCheckIntervalSeconds: desiredTargetGroupInput.HealthCheckIntervalSeconds,
+			HealthCheckTimeoutSeconds:  desiredTargetGroupInput.HealthCheckTimeoutSeconds,
+			HealthyThresholdCount:      desiredTargetGroupInput.HealthyThresholdCount,
+			UnhealthyThresholdCount:    desiredTargetGroupInput.UnhealthyThresholdCount,
+		}
+		_, err := svc.ELB.ModifyTargetGroup(modifyTargetGroup)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if targetGroupArn == nil {
+		targetGroupArn = describeTargetGroupOutput.TargetGroups[0].TargetGroupArn
+	}
+
+	if len(controlPlaneIPs) != 0 {
+		targetDescription := []*elbv2.TargetDescription{}
+		for _, controlPlaneIP := range controlPlaneIPs {
+			targetDescription = append(targetDescription, &elbv2.TargetDescription{
+				AvailabilityZone: aws.String(string("all")),
+				Id:               aws.String(controlPlaneIP),
+				Port:             aws.Int64(int64(6443)),
+			})
+		}
+
+		registerTargetInput := &elbv2.RegisterTargetsInput{
+			TargetGroupArn: targetGroupArn,
+			Targets:        targetDescription,
+		}
+		_, err = svc.ELB.RegisterTargets(registerTargetInput)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return targetGroupArn, nil
+}
+
+func convertTargetGroupOutputToInput(output *elbv2.TargetGroup) *elbv2.CreateTargetGroupInput {
```

Comment:
Hi @yastij,

Unless it's necessary to pass by reference, it's generally advised to pass by copy.